tests/webTables.spec.ts

```diff
@@ -1,222 +1,133 @@
-        const rowData = row.getByRole('cell')
-
-        // Make sure the row has cells
-        if((await rowData.all()).length > 0){
-            // Locate the desired phone number...
-            if (await rowData.nth(3).textContent() === phoneNumber) {
-                // Capture the pet name...
-                const petName = await rowData.last().textContent()
-                
-                // ...and click on the owner's name. There is only one link in the row, and that is the owner's name.
-                await rowData.getByRole('link').click()
-
-                // Wait for all the tables to load
-                await page.waitForTimeout(timeout)
-
-                /* 3. On the Owner Information page, add the assertion that "Telephone" value in the Onner Information card is "6085552765" */
-                // Validate that the phone number is listed correctly
-                await expect(page.getByRole('cell', {name: 'Telephone'}).locator('..').getByRole('cell').last()).toHaveText(phoneNumber)
-
-                /* 4. Add the assertion that Pet Name in the Owner Information card matches the name extracted from the page on the step 2 */
-                // Find table row "name" to validate the correct pet name appears
-                // Assumes there is only one pet in the Owner Information page
-                await expect(page.locator('dt', {hasText: 'Name'}).locator('..').locator('dd').first()).toHaveText(petName)
+test.describe('Validate Owner Information', () => {
+    
+    test.beforeEach(async ({ page }) => {
+        /* 1. Select the OWNERS menu item in the navigation bar and then select "Search" from the drop-down menu */
+        // Navigate to the owners table
+        await page.getByText('Owners').click()
+        await page.getByText('Search').click()
+    })
+    test('Validate the pet name city of the owner', async ({ page }) => {
+        /* 2. In the list of Owners, locate the owner by the name "Jeff Black". 
+        Add the assertions that this owner is from the city of "Monona" and he has a pet with a name "Lucky" */
+        // Find Jeff Black in the table 
+        const row = page.getByRole('row', {name: 'Jeff Black'})
+    
+        // Validate the owner is from Monona
+        await expect(row.locator('td').nth(2)).toHaveText('Monona')
+    
+        // Validate the owner has a pet named Lucky
+        await expect(row.locator('td').nth(4)).toHaveText('Lucky')
+    
+    });
+    
+    test('Validate owners count of the Madison city', async ({ page }) => {
+        /* 2. In the list of Owners, locate all owners who live in the city of "Madison". 
+        Add the assertion that the total number of owners should be 4 */
+        
+        // Validate there are only 4 entries of 'Madison' in the city column
+        await expect(page.getByRole('row', {name: 'Madison'})
+            // Filtering specifically for the City column
+            .filter({has: page.getByRole('cell').nth(2)})).toHaveCount(4)
```

Comment:
Simplify, Simplify, Simplify.
Why do you need get cells to get the count of how many owners in the Madison city? You just need the count or rows that has this city. By getting rows, you ll get owners.
So it's just `await expect(page.getByRole('row', {name: 'Madison'}).toHaveCount(4)` and you are golden